### PR TITLE
Update cilium-cli to v0.9.0

### PR DESCRIPTION
**Pull request by tklauser**
See https://github.com/cilium/cilium-cli/releases/tag/v0.9.0 for details.

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]

**Comment by nbusseneau** [comment]
GKE https://github.com/cilium/cilium/actions/runs/1208605092 and multicluster https://github.com/cilium/cilium/actions/runs/1208605086 tests both hit https://github.com/cilium/cilium-cli/issues/367.

Since this improves the CI state thanks to the ipcache fix, this is an exception to the zero-flake strategy and we are allowed to merge this once the `DO NOT MERGE` commit has been removed.

**Comment by tklauser** [comment]
> GKE https://github.com/cilium/cilium/actions/runs/1208605092 and multicluster https://github.com/cilium/cilium/actions/runs/1208605086 tests both hit [cilium/cilium-cli#367](https://github.com/cilium/cilium-cli/issues/367).
> 
> Since this improves the CI state thanks to the ipcache fix, this is an exception to the zero-flake strategy and we are allowed to merge this once the `DO NOT MERGE` commit has been removed.

Dropped `DO NOT MERGE` commit and marking as ready to merge.

**Review by joestringer — APPROVED** [review]

**Pull request merged**
